platforms/core-execution/hashing/src/main/java/org/gradle/internal/hash/HashCode.java

```diff
@@ -141,6 +159,13 @@ private static StringBuilder toStringBuilder(int capacity, byte[] bytes) {
         return sb;
     }
 
+    /**
+     * Returns a human friendly representation of the hash.
+     * <p>
+     * The compact string retains all the information of the hash.
+     * However, it cannot be used to restore the hash via {@link #fromString(String)}.
+     * For encoding/decoding use {@link #toString()}.
+     */
```

Comment:
> I tried encoding using toCompactString() and then decoding using HashCode.fromString() and it fails.

That's not the intention, and it can't work. As I said, we have compact string because some older things rely on it. All new stuff uses the `toString()` representation.

Reply:
Oh, I misunderstood. You want to eventually "retire" the compact string method.

Until we do, I think we should come up with a clean javadoc, that guides the users of this method, or defers them from using it more often.

That's where your understanding of the usages is better than mine, and it would be great, if you could provide a short description that fits the reality.